pkg/apis/certmanager/validation/certificate.go

```diff
@@ -104,6 +108,20 @@ func validateACMEConfigForAllDNSNames(a *v1alpha1.CertificateSpec, fldPath *fiel
 	return el
 }
 
+func validateIPAddresses(a *v1alpha1.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	if len(a.IPAddresses) <= 0 {
+		return nil
+	}
+	el := field.ErrorList{}
+	for i, d := range a.IPAddresses {
+		ip := net.ParseIP(d)
+		if ip == nil {
+			el = append(el, field.Invalid(fldPath.Child("ipAddresses").Index(i), d, "Invalid IP Address"))
+		}
+	}
+	return el
+}
+
```

Comment:
ACME does not allow users to request certificates with the IPAddresses field set, so we need to add additional validation to disallow certificates that reference ACME issuers having the 'ipAddresses' field set - you can see an example of a few other fields here: https://github.com/jetstack/cert-manager/blob/master/pkg/apis/certmanager/validation/certificate_for_issuer.go#L51